### PR TITLE
More tags

### PR DESCRIPTION
**Pull request by Fortifier42**
Added `<el@element.contains_all[<element>|...]>` & `<el@element.contains_all_text[<element>|...]>`
Added `<el@element.contains_all_case_sensitive[<element>|...]>` & `<el@element.contains_all_case_sensitive_text[<element>|...]>`
Added `<li@list.randomize>`

### CODE REVIEW

File: src/main/java/net/aufdemrand/denizencore/objects/dList.java [hunk 1]
```diff
@@ -309,6 +309,27 @@ public String identifySimple() {
...
+            public String run(Attribute attribute, dObject object) {
+                dList list = (dList) object;
+                if (((dList) object).isEmpty()) {
+                    return new Element("li@").getAttribute(attribute.fulfill(1));
+                }
+                Collections.shuffle(list, new Random(System.currentTimeMillis()));
+                return new Element(list.toString()).getAttribute(attribute.fulfill(1));
+            }
+        });
+        registerTag("randomized", registeredTags.get("randomize"));
```
Comment (mcmonkey4eva) [inline src/main/java/net/aufdemrand/denizencore/objects/dList.java]: This whole chunk is useless.

EDIT: And wrongly implemented anyway

Comment (Kronyn) [inline src/main/java/net/aufdemrand/denizencore/objects/dList.java]: It.. works...

Comment (Fortifier42) [inline src/main/java/net/aufdemrand/denizencore/objects/dList.java]: It works, and is more efficient than list.random?

Comment (mcmonkey4eva) [inline src/main/java/net/aufdemrand/denizencore/objects/dList.java]: Useless and wrong, burn it D:<

File: src/main/java/net/aufdemrand/denizencore/objects/dList.java [hunk 2]
```diff
@@ -309,6 +309,26 @@ public String identifySimple() {
...
+
+        registerTag("randomize", new TagRunnable() {
+            @Override
+            public String run(Attribute attribute, dObject object) {
+                dList list = (dList) object;
+                if (((dList) object).isEmpty()) {
+                    return new Element("li@").getAttribute(attribute.fulfill(1));
+                }
+                Collections.shuffle(list, new Random(System.currentTimeMillis()));
+                return new Element(list.toString()).getAttribute(attribute.fulfill(1));
```
Comment (mcmonkey4eva) [inline src/main/java/net/aufdemrand/denizencore/objects/dList.java:327]: This shuffles the source list, not creates a new list and shuffles that ~ bad.

File: src/main/java/net/aufdemrand/denizencore/objects/dList.java [hunk 3]
```diff
@@ -309,6 +309,26 @@ public String identifySimple() {
...
+        // returns a new list, with it's items reodered randomly.
+        // EG <li@1|2|3.randomize> could return li@1|2|3, li@2|1|3, li@3|2|1, li@2|3|1, li@3|1|2 OR li@1|3|2.
+        // -->
+
+        registerTag("randomize", new TagRunnable() {
+            @Override
+            public String run(Attribute attribute, dObject object) {
+                dList list = (dList) object;
+                if (((dList) object).isEmpty()) {
+                    return new Element("li@").getAttribute(attribute.fulfill(1));
```
Comment (mcmonkey4eva) [inline src/main/java/net/aufdemrand/denizencore/objects/dList.java:324]: Wot da fuq

File: src/main/java/net/aufdemrand/denizencore/objects/dList.java (hunk 2 above)
Comment (mcmonkey4eva) [inline src/main/java/net/aufdemrand/denizencore/objects/dList.java:327]: It also returns an ELEMENT of the list

File: src/main/java/net/aufdemrand/denizencore/objects/dList.java [hunk 4]
```diff
@@ -309,6 +309,26 @@ public String identifySimple() {
...
+                }
+                Collections.shuffle(list, new Random(System.currentTimeMillis()));
```
Comment (mcmonkey4eva) [inline src/main/java/net/aufdemrand/denizencore/objects/dList.java:326]: We have a utilRandom instance to use for all randomness needs

**Pull request closed without merging**